File: Problem_Set_1/Cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

//Important functions prototypes
int get_change(string prompt);
int get_coins(int change);

int main(void)
{
    int change = get_change("Change owed: ");
    int coins = get_coins(change);
    printf("%i\n", coins);
}

// This function make sure we get a proper value for the change, and round it
int get_change(string prompt)
{
    float change_float;
    int change;
    do
    {
        change_float = get_float("%s", prompt);
        change = round(change_float * 100);
    }
    while (change < 0);
    return change;
}

// This functions return the proper amount of coins owed
int get_coins(int change)
{
    int coins = 0;
    while (change >= 25)
    {
        change -= 25;
        coins++;
    }
    while (change >= 10)
    {
        change -= 10;
        coins++;
    }
    while (change >= 5)
    {
        change -= 5;
        coins++;
    }
    coins += change;
    return coins;
}

File: Problem_Set_6/cash/cash.py
from cs50 import *
# Get the value from the user
while True:
    value = get_float("Change owed: ")
    if (int(value) >= 0):
        break

# Compute the total number of coins
value = int(value * 100)
coins = 0

while (value > 0):
    if (value >= 25):
        coins += (value // 25)
        value %= 25
    elif (value >= 10):
        coins += (value // 10)
        value %= 10
    elif (value >= 5):
        coins += (value // 5)
        value %= 5
    elif (value >= 1):
        coins += (value // 1)
        value %= 1

# Print out the number of coins
print(coins)
